types/utils/utils.go

```diff
@@ -0,0 +1,32 @@
+package utils
+
+import (
+	"errors"
+	"github.com/holiman/uint256"
+	"math/big"
+	"strings"
+)
+
+func HexStringToBigInt(input string) (*big.Int, error) {
+	numPart := strings.TrimPrefix(input, "0x")
+	trimmed := strings.TrimLeft(numPart, "0")
```

Comment:
If I'm not making a mistake, `TrimLeft` trims all the characters to the end. 
This will cause an issue for `0x0` , ... `0x0..0`  kind of input since `uint256.FromHex` generates error for `0x`. 
It would be great if you can check it and guard it if required. 
Same issue also in `HexStringToUint64` function